data/basics/userBalance.go

```diff
@@ -479,20 +491,36 @@ func (u AccountData) MinBalance(proto *config.ConsensusParams) (res MicroAlgos)
 	return res
 }
 
-// VotingStake returns the amount of MicroAlgos associated with the user's account
-// for the purpose of participating in the Algorand protocol.  It assumes the
-// caller has already updated rewards appropriately using WithUpdatedRewards().
-func (u AccountData) VotingStake() MicroAlgos {
+// OnlineAccountData returns subset of AccountData as OnlineAccountData data structure.
+// Account is expected to be Online otherwise its stake is cleared out
+func (u AccountData) OnlineAccountData() OnlineAccountData {
```

Comment:
The code here suggest that you could have a situation where you have a non-online account which has any of the VoteID/SelectionID/VoteFirstValid/VoteLastValid/VoteKeyDilution non-zero, which isn't correct.
If Status is online, then all these fields are valid. Otherwise, they are all empty.

Reply:
@cce opinion?